mechanicalsoup/form.py

```diff
@@ -256,10 +256,10 @@ def new_control(self, type, name, value, **kwargs):
         self.form.append(control)
         return control
 
-    def choose_submit(self, el):
```

Comment:
Note that you're potentially breaking backward compatibility for people using `choose_submit(el=foo)`. `choose_submit` was introduced relatively recently so I don't think it has so many users and we can hope that nobody wrote this `el=foo`, so I'm OK with this.

Reply:
Why is it so painful to do the right thing! ;)

I agree that most people would have written `choose_submit(foo)` instead of `choose_submit(el=foo)`, and `el` is a painfully out-of-place variable name, but your instinct to preserve backwards compatibility is a good one. For now, I'll separate this change out into another PR so we can think on it for a bit longer.

Reply:
At worse, you can keep `el=None` as argument, and do stg like

    if el is not None:
        print("el is deprecated, please use submit=... instead")
        if submit is not None:
            raise ...
        submit = el

We'd really need that if we were a 10,000,000 users lib. Probably overkill here. Actually, if someone wrote `el=...` somewhere in a script, then this script doesn't work but it raises a reasonable error and should be straightforward to fix.

Backward compatibility is painful for us, but it's really annoying to run a 2-year old script and notice it doesn't work anymore.